src/meta.rs

```diff
@@ -5,9 +5,20 @@ pub struct Meta {
     pub universe: Vec<AssetMeta>,
 }
 
+#[derive(Deserialize, Debug, Clone)]
```

Comment:
Make this type accurate and have better names:

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct SpotMeta {
    universe: Vec<SpotInfo>,
    tokens: Vec<TokenInfo>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
struct SpotInfo {
    tokens: [usize; 2],
    name: String,
    index: usize,
    is_canonical: bool,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
struct TokenInfo {
    name: String,
    sz_decimals: u8,
    wei_decimals: u8,
    index: usize,
    token_id: H128,
    is_canonical: bool,
}

I also don't think you need to implement clone, but feel free to add it if it's helpful

Reply:
Done. I've added Clone to stick to the standard set by the AssetMeta struct. Just seems like a good addition to avoid future problems maybe?